src/main/java/com/minecolonies/coremod/client/gui/townhall/WindowPermissionsPage.java

```diff
@@ -451,7 +450,12 @@ private void trigger(@NotNull final Button button)
 
         final IPermissions permissions = building.getColony().getPermissions();
         final PlayerEntity playerEntity = Minecraft.getInstance().player;
-        final boolean enable = !LanguageHandler.format(COM_MINECOLONIES_COREMOD_GUI_WORKERHUTS_RETRIEVE_ON).equals(button.getTextAsString());
+        String key = button.getTextAsString();
+        if (button.getText() instanceof TranslationTextComponent)
```

Comment:
could be oneliner using ternary (also at other places - search for casting to TranslationTextComponent to find them)

Reply:
There were only 3 occurences of this, a couple of them were using `I18n.exists` which I didn't know how to replace, so I left them for now.